lib/idl/check_inputs.pro

```diff
@@ -108,6 +108,7 @@ PRO check_inputs, inputs
                                   'seed', -1, $
                                   'stark_components', 0, $
                                   'neutrals_file',neutrals_file)
+    if (total(tag_names(inputs) EQ 'OUTPUT_NEUTRAL_RESERVOIR') EQ 0) then inputs=create_struct(inputs,'output_neutral_reservoir',1)
```

Comment:
@alvin-garcia @lstagner Not an idl expert... Should the items above, i.e seed, stark_components that have default values etc... be set in a similar way to what I m doing here? Otherwise if those values are already in inputs, then check_inputs will bomb...

Reply:
The options added here are mainly for developers so they are not assumed to be in the prefida inputs structure initially. Strictly speaking your way is better but for now can you change it to match how the other ones are set.

Reply:
Alternatively you can create a new procedure (defined below) in a new file and use that for all the defaults
```idl
PRO set_struct_default, struct, name, data
    IF (total(tag_names(struct) eq strupcase(name)) eq 0) THEN BEGIN
        struct = create_struct(struct,name,data)
    ENDIF
END
```

Reply:
@shaunhaskey, your code is good. Please implement @lstagner's suggestion in `preprocessing.py` as well. Then, execute `run_tests, '/some/path/you/like'` in IDL and `run_tests.py /some/path/you/like'` from the command line. This is an easy way to check nothing bombed lol

Reply:
I ran into an problem with run_tests.pro, which I think was there before I started. @alvin-garcia  or @lstagner Please can you check to see if the existing test works...?

```
(/u/shaskey/conda) [shaskey@sunfire07 test]$ export FIDASIM_DIR=/u/shaskey/FIDASIM_MS_centos7/
(/u/shaskey/conda) [shaskey@sunfire07 test]$ 
(/u/shaskey/conda) [shaskey@sunfire07 test]$ export PATH=$FIDASIM_DIR/deps/hdf5/bin:$FIDASIM_DIR/lib/scripts:$PATH
(/u/shaskey/conda) [shaskey@sunfire07 test]$ export PATH=$FIDASIM_DIR/deps/efit:$FIDASIM_DIR/test:$PATH
(/u/shaskey/conda) [shaskey@sunfire07 test]$ export IDL_PATH="+$FIDASIM_DIR:$IDL_PATH:<IDL_DEFAULT>"
(/u/shaskey/conda) [shaskey@sunfire07 test]$ export PYTHONPATH=$FIDASIM_DIR/lib/python:$PYTHONPATH
(/u/shaskey/conda) [shaskey@sunfire07 test]$ pwd
/u/shaskey/FIDASIM_MS_centos7/test
(/u/shaskey/conda) [shaskey@sunfire07 test]$ idl
IDL 8.8.0 (linux x86_64 m64).
(c) 2020, Harris Geospatial Solutions, Inc.

Licensed for use by: Princeton Plasma Physics Lab (Main)
License: 100897
IDL> run_tests,"/p/fida/shaskey/fida_ruN_test_idl/"
% Compiled module: RUN_TESTS.
% Compiled module: GET_FIDASIM_DIR.
% Compiled module: SOURCE_FILE.
% Compiled module: TEST_BEAM.
% Compiled module: RZ_GRID.
% Loaded DLM: LAPACK.
% Compiled module: READ_GEQDSK.
% Compiled module: READG.
% Compiled module: EFIT_READ.
% Compiled module: EFIT_FILENAME_PARSE.
  >>> in READG_FILE, reading info.file /u/shaskey/FIDASIM_MS_centos7/test/g000001.01000
% Type conversion error: Unable to convert given STRING to Long.
% Detected at: READG_FILE        461 /u/shaskey/FIDASIM_MS_centos7/deps/efit/readg.pro
% Compiled module: STR_SEP.
% READG_FILE: Trouble determining exact time.  Filename gives: 1000.0000  but file contains: 1.00000
% Compiled module: DT_NICENUMBER.
% Compiled module: NINT.
% Compiled module: RHO_RZ.
% Compiled module: LINSPACE.
% Compiled module: FLUXMAP.
% Compiled module: SPLINE.
% Compiled module: QSIMPV.
% Compiled module: CONTOUR_PSI.
% Compiled module: METRIC.
% Compiled module: DERIV.
% Compiled module: IN_VESSEL.
% Compiled module: READ_NUBEAM.
% Compiled module: READ_NCDF.
% Loaded DLM: NCDF.
Species: D_NBI
Energy min/max:      0.00000      121.681
Pitch  min/max:     -1.00000      1.00000
Ntotal in phase space:    1.2136334e+19
% Loaded DLM: QHULL.
% Compiled module: ARRAY_INDICES.
% Compiled module: INTERPOL.
% Compiled module: TEST_CHORDS.
% Compiled module: TEST_NPA.
% Compiled module: TEST_PROFILES.
% REBIN: Result dimensions must be integer factor of original dimensions
% Execution halted at: TEST_PROFILES      31 /u/shaskey/FIDASIM_MS_centos7/test/test_profiles.pro
%                      RUN_TESTS          54 /u/shaskey/FIDASIM_MS_centos7/test/run_tests.pro
%                      $MAIN$          
IDL>
```
The error is happening at a line which might be having issues with multispecies changes...:
```
    deni=rebin(denimp,nthermal,nr,nz,grid.nphi)
```